game/src/edit/roads.rs

```diff
@@ -1,84 +1,461 @@
+    .build(ctx)
+}
+
+fn make_main_panel(
+    ctx: &mut EventCtx,
+    app: &App,
+    road: &Road,
+    current_lane: Option<LaneID>,
+) -> Panel {
+    let map = &app.primary.map;
+
+    let modify_lane = if let Some(l) = current_lane {
+        let lane = map.get_l(l);
+        let idx = road.offset(l);
+
+        Widget::row(vec![
+            ctx.style()
+                .btn_plain
+                .icon("system/assets/tools/trash.svg")
+                .disabled(road.lanes_ltr().len() == 1)
+                .build_widget(ctx, "delete lane"),
+            ctx.style()
+                .btn_plain
+                .text("flip direction")
+                .disabled(!can_reverse(lane.lane_type))
+                .build_def(ctx),
+            Line("Width").secondary().into_widget(ctx),
+            Widget::dropdown(ctx, "width", lane.width, width_choices(app, l)),
+            ctx.style()
+                .btn_plain
+                .text("<")
+                .disabled(idx == 0)
+                .build_widget(ctx, "move lane left"),
+            ctx.style()
+                .btn_plain
+                .text(">")
+                .disabled(idx == road.lanes_ltr().len() - 1)
+                .build_widget(ctx, "move lane right"),
+        ])
+    } else {
+        Widget::nothing()
+    };
+    let current_lt = current_lane.map(|l| map.get_l(l).lane_type);
+
+    let mut available_lane_types_row = vec![
+        LaneType::Driving,
+        LaneType::Biking,
+        LaneType::Bus,
+        LaneType::Parking,
+        LaneType::Construction,
+        LaneType::Sidewalk,
     ]
+    .into_iter()
+    .map(|lt| {
+        ctx.style()
+            .btn_plain
+            .icon(lane_type_to_icon(lt).unwrap())
+            .disabled(Some(lt) == current_lt)
+            .build_widget(
+                ctx,
+                format!(
+                    "{} {}",
+                    if current_lane.is_some() {
+                        "change to"
+                    } else {
+                        "add"
+                    },
+                    lt.short_name()
+                ),
+            )
+    })
+    .collect::<Vec<Widget>>();
+    if current_lane.is_some() {
+        available_lane_types_row.insert(0, "change to".text_widget(ctx));
+    } else {
+        available_lane_types_row.insert(0, "add new".text_widget(ctx));
+    }
+    let available_lane_types_row = Widget::row(available_lane_types_row);
+
+    let mut current_lanes_ltr = Vec::new();
+    for (id, _, lt) in road.lanes_ltr() {
+        // TODO Add direction arrow sometimes
+        current_lanes_ltr.push(
+            ctx.style()
+                .btn_plain
+                .icon(lane_type_to_icon(lt).unwrap())
+                .disabled(Some(id) == current_lane)
+                .build_widget(ctx, format!("modify {}", id)),
+        );
+    }
+    let current_lanes_ltr = Widget::row(current_lanes_ltr);
+
+    let road_settings = Widget::row(vec![
```

Comment:
I know it's under construction and likely to change, but I think vertically centering labels with their controls and ensuring content isn't crammed up against each other should be a relatively easy way to fix some egregious layout:

<img width="224" alt="Screen Shot 2021-04-21 at 5 26 37 PM" src="https://user-images.githubusercontent.com/217057/115638270-e7bbc900-a2c6-11eb-9d0c-19140e9a89ab.png">

Reply:
Done
![Screenshot from 2021-04-22 07-48-38](https://user-images.githubusercontent.com/1664407/115735235-2ab98300-a33f-11eb-9f97-e9ec6696c81d.png)